app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/SpatialCache.kt

```diff
@@ -0,0 +1,138 @@
+package de.westnordost.streetcomplete.data.osm.osmquests
+
+import android.util.Log
+import de.westnordost.streetcomplete.data.download.tiles.TilePos
+import de.westnordost.streetcomplete.data.download.tiles.enclosingTilePos
+import de.westnordost.streetcomplete.data.download.tiles.enclosingTilesRect
+import de.westnordost.streetcomplete.data.download.tiles.minTileRect
+import de.westnordost.streetcomplete.data.osm.mapdata.BoundingBox
+import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
+import de.westnordost.streetcomplete.util.math.contains
+import de.westnordost.streetcomplete.util.math.isCompletelyInside
+
+// inspired by Android LruCache
+// fetch needs get data from db and fill other caches, like questKey -> Quest
+// onRemoveTile needs to clean other caches, like questKey -> Quest
+// careful: put does not actually put key into cache if tile is not cached -> update db blocking if key was not put
+// TODO: synchronized stuff? didn't get related crash so far...
+// TODO: tile zoom as parameter? could smaller tiles help for mapDataController?
```

Comment:
yes, should be parameter. No reason to hardcode this from the point of view of this class. Whether smaller tiles will help, I would say, **probably** yes in case of the mapdatacontroller because oftentimes bbox-queries are not only done per-displayed-tile but also e.g. surrounding a quest. But this can be optimized later.

Reply:
ok. I changed the tileZoom to a parameter